Soho vibes for RStudio

eric-hunt opened the pull request:
Hey @gadenbuie,
Here's another pull request for the Rosé Pine themes. There are two dark variants (the base theme, and "Moon") and one light variant ("Dawn"). I kind of copied the way the Material themes are built for this one to recursively go through the variants and update the palette accordingly.
![base](https://github.com/user-attachments/assets/edc576e1-4cd5-4211-9511-60521d3ad25d)
![moon](https://github.com/user-attachments/assets/93b9bce4-c3af-4131-bfaf-e18c2fda7bd6)
![dawn](https://github.com/user-attachments/assets/ffd26f79-1ae6-4493-ab63-8cfc4226d8a3)
Also, I updated a link for the Embark theme in the palette list file because I realized after deleting the merged branch, the reference now pointed to nowhere.

eric-hunt committed:
Added official palette colors

eric-hunt committed:
Added build/install function based on rsthemes::rstheme()

eric-hunt committed:
Added sass export

eric-hunt committed:
Changed Boolean and message color

eric-hunt committed:
Ran make.R

eric-hunt committed:
Altered export function and added rstheme file export

eric-hunt committed:
Collective terminal, code, and UI color updates from eric-hunt/rosepine.rstheme

eric-hunt committed:
Removed redundant make commands

eric-hunt committed:
Ran make.R

eric-hunt committed:
Revert changes in fc9d1fd

eric-hunt committed:
Added bullet to palette list

eric-hunt committed:
Added bullet to NEWS

gadenbuie committed:
Merge branch 'main' into rose-pine

gadenbuie commented on inst/templates/rosepine.R:
```diff
@@ -0,0 +1,500 @@
+      .gwt-TabLayoutPanelTabInner .rstheme_tabLayoutCenter {
+        box-shadow: 0 3px 0 $cyan inset;
+        border-radius: 0 !important;
+
+        .gwt-Label {
+          \\ font-weight: 600;
+          \\ color: $fg;
+        }
+      }
```
Let's just delete these lines... 

```suggestion
```

or we need to update the number of slashes (because it's a character string in R)

```suggestion

        .gwt-Label {
          \\\\ font-weight: 600;
          \\\\ color: $fg;
        }
      }
```

gadenbuie commented on inst/templates/rosepine.R:
```diff
@@ -0,0 +1,500 @@
+  lighten_factor <- 5
+  darken_factor <- 5
+  .lighten <- function(nm, by = lighten_factor) {
+    paste0(
+      "lighten($",
+      nm,
+      ", ",
+      by,
+      "%)"
+    )
```
```suggestion
    sprintf("lighten($%s, %s%%)", nm, by)
```

eric-hunt replied:
Way cleaner, thanks.

gadenbuie commented on inst/templates/rosepine.R:
```diff
@@ -0,0 +1,500 @@
+    )
+  }
+  .darken <- function(nm, by = darken_factor) {
+    paste0(
+      "darken($",
+      nm,
+      ", ",
+      by,
+      "%)"
+    )
```
```suggestion
    sprintf("darken($%s, %s%%)", nm, by)
```

gadenbuie commented on inst/templates/rosepine.R:
```diff
@@ -0,0 +1,500 @@
+#' defaults if necessary
+#'
+#' @export
+rosepine_rstheme <- function(
+    variant = "base",
+    apply = FALSE,
+    as_sass = FALSE,
+    ...) {
+  variant.choices <- list("base" = NULL, "moon" = "Moon", "dawn" = "Dawn")
+  stopifnot(
+    "Variant must be 'base', 'moon', or 'dawn'." =
+      tolower(variant) %in% names(variant.choices)
+  )
+  variant <- do.call(switch, c(tolower(variant), variant.choices))
```
This code gets you to the same place but I think it's easier to follow. It's also okay to use rlang here even though it's not a package dependency, this script is only run in development.

```suggestion
rosepine_rstheme <- function(
  variant = c("base", "moon", "dawn"),
  apply = FALSE,
  as_sass = FALSE,
  ...
) {
  rlang::arg_match(variant)
  variant <- switch(variant, moon = "Moon", dawn = "Dawn")
```

gadenbuie commented on inst/templates/rosepine.R:
```diff
@@ -0,0 +1,500 @@
+
+  theme_dark <- !identical(variant, "Dawn")
+
+  rp_pal <- get(paste0(
+    "rose_pine",
+    if (!is.null(variant)) "_",
+    tolower(variant)
+  ))
```
Similarly, this code is less clever but easier to read and maintain.

```suggestion
  rp_pal <- switch(
    variant,
    Dawn = rose_pine_dawn,
    Moon = rose_pine_moon,
    rose_pine
  )
```

gadenbuie reviewed (commented):
Looks great, thank you! I have just a few small style and housekeeping comments

eric-hunt committed:
Remove tab panel font weight and color scss

Incorrect escaping when passed as R string

Co-authored-by: Garrick Aden-Buie <garrick@adenbuie.com>

eric-hunt committed:
Cleaner lighten helper function

Co-authored-by: Garrick Aden-Buie <garrick@adenbuie.com>

eric-hunt committed:
Cleaner transparentize helper function

Co-authored-by: Garrick Aden-Buie <garrick@adenbuie.com>

eric-hunt committed:
Cleaner darken helper function

Co-authored-by: Garrick Aden-Buie <garrick@adenbuie.com>

eric-hunt committed:
Let make.R build the .rstheme files from sass

eric-hunt committed:
More succinct handling of variant argument

Co-authored-by: Garrick Aden-Buie <garrick@adenbuie.com>

eric-hunt committed:
Simpler palette selection from variant argument

Co-authored-by: Garrick Aden-Buie <garrick@adenbuie.com>

eric-hunt committed:
Rename template file

eric-hunt committed:
Fixed switch and naming error caused by 8075780

eric-hunt committed:
Fixed number formating for transparentize helper in 5f93fce

eric-hunt committed:
Deleted too many brackets in f3589d0

eric-hunt committed:
Enacted same change for Embark as mentioned in #102

eric-hunt committed:
Sourced make.R

eric-hunt commented:
Thank you for the input! I have made the changed you suggested and renamed the file.

I had to change what you suggested for handling the `variant` argument a little bit because the `switch()` for the palette was throwing an error if the `variant` argument didn't explicitly return `"base"`. Also I had to change how I programmatically named the files and theme as I was relying on `is.null()` with the `variant` argument.

I also went ahead and removed those ill-escaped scss lines from the embark.R template as well so they didn't live on kicking around in that theme.

I re-sourced make.R and it seems like everything worked, so if those changed I made to your suggestions look okay to you, I think this is good to go.

gadenbuie committed:
chore: simplify output file logic

gadenbuie commented on inst/templates/rose-pine.R:
```diff
@@ -167,42 +167,18 @@ rosepine_rstheme <- function(
-      )
-    }
-  }
+  theme_file <- paste0(
```
@eric-hunt I simplified this logic a bit. The new version should do the same thing, but let me know if I missed a detail somewhere.

eric-hunt replied:
That's so much simpler. 😆 Thanks!

gadenbuie reviewed (approved):
Thanks again @eric-hunt!
